src/sensors/can/C620.cpp

```diff
@@ -0,0 +1,81 @@
+#include "C620.hpp"
+
+void C620::init() {
+    zero_motor();
+}
+
+int C620::read(CAN_message_t& msg) {
+    // early return if the message ID does not match
+    if (msg.id != m_base_id + m_id) return 0;
+    
+    // set m_input from msg
+    memcpy(&m_input, &msg, sizeof(CAN_message_t));
+
+    // fill out the motor state buffer
+    int16_t torque = (m_input.buf[4] << 8) | m_input.buf[5];
+    m_state.torque = (float)torque / m_max_torque;
+    
+    int16_t rpm = (m_input.buf[2] << 8) | m_input.buf[3];
+    float rad_per_sec = rpm * ((2 * PI) / 60);
+    m_state.speed = rad_per_sec;
+    m_state.position = (m_input.buf[0] << 8) | m_input.buf[1];
+    m_state.temperature = m_input.buf[6];
+
+    return 1;
+}
+
+int C620::write(CAN_message_t& msg) const {
+    // set ID
+    msg.id = m_output.id;
+
+    // set only the buffer bytes that correspond to this motor
+    // get the per-struct motor ID
+    uint8_t motor_id = (m_id - 1) % 4;
+
+    // fill in the output array
+    msg.buf[motor_id * 2] = m_output.buf[motor_id * 2];         // upper byte
+    msg.buf[motor_id * 2 + 1] = m_output.buf[motor_id * 2 + 1]; // lower byte
```

Comment:
again, description of buffer data would be helpful (apologies for mentioning this *every time*, but I want to just note all the places where I feel it can be more clear)